FirebaseAuth/Sources/Swift/Auth/Auth.swift

```diff
@@ -168,7 +168,7 @@ extension Auth: AuthInterop {
    @return The `Auth` instance associated with the given app.
    */
   @objc open class func auth(app: FirebaseApp) -> Auth {
-    return ComponentType<AuthProvider>.instance(for: AuthProvider.self, in: app.container) as! Auth
+    return ComponentType<AuthProvider>.instance(for: AuthProvider.self, in: app.container).auth()
```

Comment:
Now the type can be right here, since we return an AuthComponent instead of an Auth below.